id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/BiometricExtractionServiceImpl.java

```diff
@@ -82,13 +83,21 @@ public CompletableFuture<List<BIR>> extractTemplate(String uinHash, String fileN
 			String extractionType, String extractionFormat, List<BIR> birsForModality) throws IdRepoAppException {
 		try {
 			String extractionFileName = fileName.split("\\.")[0] + DOT + getModalityForFormat(extractionType) + DOT + extractionFormat;
+			Map<String, String> formatFlag = Map.of(getFormatFlag(extractionType), extractionFormat);
 			// TODO need to remove AmazonS3Exception handling
 			try {
 				if (objectStoreHelper.biometricObjectExists(uinHash, extractionFileName)) {
 					mosipLogger.info(IdRepoSecurityManager.getUser(), this.getClass().getSimpleName(), EXTRACT_TEMPLATE,
 							"RETURNING EXISTING EXTRACTED BIOMETRICS FOR FORMAT: " + extractionType +" : "+ extractionFormat);
 					byte[] xmlBytes = objectStoreHelper.getBiometricObject(uinHash, extractionFileName);
 					List<BIR> existingBirs = cbeffUtil.convertBIRTypeToBIR(cbeffUtil.getBIRDataFromXML(xmlBytes));

```

Comment:
If the xml is invalid, it may also throw exception with `cbeffUtil.getBIRDataFromXML` So please handle the same and in that case also the bio-extraction needs to be called.

Reply:
Changes done